shared/src/main/scala/org/atnos/eff/IntoPoly.scala

```diff
@@ -26,7 +26,7 @@ trait IntoPolyLower1 extends IntoPolyLower2 {
     new IntoPoly[NoFx, R] {
       def apply[A](e: Eff[NoFx, A]) =
         e match {
-          case Pure(a, last) => pure[R, A](a).addLast(last.interpret(apply))
+          case Pure(_, _) => e.asInstanceOf[Eff[R, A]]
```

Comment:
I was not sure that the cast was safe here, but I think it is because `last` can only be an `Option[Eval[Pure[Unit]]`.